components/Collections/src/AdvancedCollectionView/AdvancedCollectionView.cs

```diff
@@ -13,6 +13,9 @@ namespace CommunityToolkit.WinUI.Collections;
 /// <summary>
 /// A collection view implementation that supports filtering, sorting and incremental loading
 /// </summary>
+#if NET8_0_OR_GREATER
+[System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("Item sorting uses reflection to get property types and may not be AOT compatible.")]
```

Comment:
Shouldn't the item sorting uses work without problems with AOT? At least with UWP with .NET native there are no problems, or are there changes in the implementation?

Reply:
[[docs](https://learn.microsoft.com/en-us/windows/uwp/dotnet-native/net-native-and-compilation#just-in-time-compilation)] UWP with .NET Native uses the default runtime directive for libraries: 
```
<Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
  <Application>
    <Assembly Name="*Application*" Dynamic="Required All" />
  </Application>
</Directives>
```
> This enables all the types, as well as all their members, in all the assemblies in your app package for reflection and dynamic invocation. However, it does not enable reflection or dynamic activation of types in .NET Framework Class Library assemblies. In many cases, this is adequate.

For .NET 8 and NativeAOT, they use these attributes to bubble up warnings or errors to the compiler from underlying libraries to signal when AoT might not work as expected. The code change here is a simple suppression of one of these errors.

Reply:
Ah :) Thank you for the explanation.